modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/K6ClientCodegen.java

```diff
@@ -638,7 +666,19 @@ public void preprocessOpenAPI(OpenAPI openAPI) {
 
                 pathVariables.put(groupName, variables);
 
-                final HTTPParameters params = new HTTPParameters(null, null, httpParams, null, null, null, null, null,
+                // put auth medthods in header or cookie
+                for (CodegenSecurity globalAuthMethod : globalAuthMethods) {
+                    if (globalAuthMethod.isKeyInHeader) {
+                        httpParams.add(new Parameter(globalAuthMethod.keyParamName, getTemplateString(toVarName(globalAuthMethod.keyParamName))));
+                        extraParameters.add(new Parameter(toVarName(globalAuthMethod.keyParamName), globalAuthMethod.keyParamName.toUpperCase(Locale.ROOT)));
+                    }
+                    if (globalAuthMethod.isKeyInCookie) {
+                        cookieParams.add(new Parameter(globalAuthMethod.keyParamName, getTemplateString(toVarName(globalAuthMethod.keyParamName))));
+                        extraParameters.add(new Parameter(toVarName(globalAuthMethod.keyParamName), globalAuthMethod.keyParamName.toUpperCase(Locale.ROOT)));
+                    }
```

Comment:
@willie5588912 Is it either/or? Or can it be both? 🤔

Reply:
Yes, it can be in both. In some cases, we may need authentication token both in headers and cookies, and this depends on the requirement from the server side. 
As the implementation here, it gives the flexibility for authentication token in headers and cookies since they are independent and both ok if provided or not provided.

Reply:
@mostafa thanks for reviewing. I saw the thumb up so I'll merge it to get the ball rolling.